doc/tutorial/sequence_file/index.md

```diff
@@ -0,0 +1,343 @@
+You can check whether you have installed these libraries by running `cmake .` in your build directory. If `-- Optional dependency: ZLIB-x.x.x found.` is displayed on the command line then you can read/write compressed files in your programs.
+
+## Basic layout of SeqAn3 file objects
+
+Before we dive into the details, we will outline the general design of our file objects, hoping that it will make the following tutorial easier to understand.
+
+As mentioned above, our file object is a range over seqan3::record's, where a seqan3::record is basically just a std::tuple that holds the data. To identify or specialise which data is read/written and contained in the records, we use seqan3::field tags (e.g. seqan3::field::SEQ denotes sequence information). The seqan3::field tags are shared between file formats and allow for easy file conversion.
+
+The input file object has one major difference to the output file object, namely that the data types cannot be automatically deduced from the input variables but need to be specified via type traits (e.g. whether your sequence information are `dna` or `protein`). Of course we provide default traits for every format and different use cases but it is good to keep in mind that our file objects are completely generic.
+
+# Sequence Files {#section_sequence_files}
+
+Sequence files are the most generic and common biological files. Well-known formats include FASTA and FASTQ, but some may also be interested in treating SAM or BAM files as sequence files, discarding the alignment. The Sequence file abstraction supports reading four different fields:
+
+  1. seqan3::field::SEQ
+  2. seqan3::field::ID
+  3. seqan3::field::QUAL
+  4. seqan3::field::SEQ_QUAL (sequence and qualities in one range, see seqan3::qualified for details)
+
+The first three fields are retrieved by default (and in that order!). The last field may be selected to have sequence and qualities directly stored in a more memory-efficient combined container. If you select the last field you may not select seqan3::field::SEQ or seqan3::field::QUAL.
+
+# Sequence file formats
+
+### FASTA format
+
+FASTA/FASTQ files are both record-based files.
+
+A FASTA record contains the sequence id and the sequence characters. Here is an example of a FASTA file:
+
+```
+>seq1
+CCCCCCCCCCCCCCC
+>seq2
+CGATCGATC
+```
+### FASTQ format
+
+A FASTQ record contains an additional quality value for each sequence character. Here is an example of a FASTQ file:
+
+```
+@seq1
+CCCCCCCCCCCCCCC
++
+IIIIIHIIIIIIIII
+@seq2
+CGATCGATC
++
+IIIIIIIII
+```
+
+### File extensions
+
+The formerly introduced formats can be identified by the following file name extensions (This is important for automatic format detection when constructing our sequence files as you will learn in the next section).
+
+| File Format  | File Extensions   |
+| -------------|-------------------|
+| FASTA        |   `.fa`, `.fasta` |
+| FASTQ        |   `.fq`, `.fastq` |
+
```

Comment:
Maybe one sentence on how to access the acual extension list of the format and that you have the ability to change it?